### DISCUSSION TITLE
Setting cursor style in VimLeave doesn't work

### DISCUSSION THREAD

**Discussion by viveksjain** [ghostty-org/ghostty#3561] [Q&A] [answered]
I like to set a different cursor style in vim, and then reset it to my terminal default when vim exits. This works fine in iTerm. In ghostty, the cursor style changes when entering vim, but not when exiting. Here is the relevant .vimrc conf
```
silent !echo -ne '\e[2 q'  " Set cursor to box on init
autocmd VimLeave * silent !echo -ne '\e[5 q' " Reset to blinking vertical bar on exit - has no effect in ghostty
```
Using ghostty 1.0.0 on Mac. This is admittedly a pretty niche use case, but let me know if I missed anything or doing something wrong!

**Comment by 00-kat** [comment, marked as answer]
Why not try the `guicursor` option?

> Reset to blinking vertical bar on exit

Also consider checking out [shell integration](https://ghostty.org/docs/config/reference#shell-integration), which does this by default.

That snippet does not work for me in neither Ghostty, Kitty nor Foot, though.

And also (irrelevant to your issue, but I want to mention it anyway) please use `printf` instead of echo with flags, since that's not portable behaviour.

**Reply by viveksjain** [reply]
Interesting, wasn't aware of `guicursor` but that doesn't seem to do anything and it mentions only applying to gvim. Setting `t_SI` and `t_EI` in my vimrc works (I was already doing this), and shell integration is enabled, the only problem is I need to reset the cursor on vim exit and none of the approaches are working for that.
